docs/guides/javascript.md

```diff
@@ -313,3 +313,33 @@ spectral.run({description: 'Swagger is pretty cool!',}).then(results => {
 ```
 
 For more information on creating rules, read about [rulesets](../getting-started/rulesets.md).
+
+### Creating a custom format
+
+Spectral supports two core formats: `oas2` and `oas3`. Using `registerFormat` you can add support for autodetecting other formats. You might want to do this for a ruleset which is run against multiple major versions of description format like RAML v0.8 and v1.0.
+
+```js
+spectral.registerFormat('foo-bar', obj => typeof obj === 'object' && obj !== null && 'foo-bar' in obj);
+
+spectral.addRules({
+  rule1: {
+    given: '$.x',
+    formats: ['foo-bar'],
```

Comment:
Ha, we didn't discuss such a case in the original issue or I might have misunderstood something, but... if no format can be found, we run all rules regardless of their formats.
In other words, if you change it to
```suggestion
    formats: ['foo-dsadaczxcz'],
```

and 

```
const result = await spectral.run({
  'foo-baraaaa': true,
```

the rule would be executed.

Now that I think of it, I don't believe it was wise.
Changing it would be very easy though, so can do it right away.

Reply:
https://github.com/stoplightio/spectral/pull/449

Reply:
This example was not actually talking about what you are talking about currently, I just copied and pasted it from a test. But, im glad it got you thinking, because #449 is a good change. :D